### PR TITLE
most of these i just stubbed, q4 i got further, i put separate questi…

### PR DESCRIPTION
**Pull request by xgironx**
…ons into separate files so non-interfering

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @person = person
+    puts "Creating #{self.person}"   # here, `self` refers to the current instance
+    @@all.push(self)
+  end
+
+  def offer_rose
+    puts "Would you take this rose, person, in exchange for giving an old beggar woman shelter from the bitter cold?"
+    return "#{@person.capitalize}"
+  end
+
```
Comment (superbuggy) [inline app.rb:64]: all you need for this question is a offer_rose method that takes an argument, and not a person class

File: app_q2.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                castle: {
+                  num_rooms: 47,
+                  residents: "Robby Benson",
+                  guests: []
+                }
+              }
+
+              residents.
+
+              town[:residents]
```
Comment (superbuggy) [inline app_q2.rb:32]:
```rb
town[:residents].delete("Belle")
town[:castle][:guests].push("Belle")
```

File: app_q4.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              class Lion extends Animal
+                attr_accessor :name
+                @@all = []
+                king = false
+
+                def initialize(name)
+                  @@pack
+                  @name = name
+                  puts "Creating #{self.name}"   # here, `self` refers to the current instance
+                  @@all.push(self)
```
Comment (superbuggy) [inline app_q4.rb:53]:
```rb
if @name == "Simba"
  @king = true
end
```

**Comment by superbuggy** [comment]
If you find yourself short on time, answer any terminology questions first then come back to the coding questions

**Comment by xgironx** [comment]
James, I just got checked back in all but the last one, the Sinatra browser greeting.  Jay

**Pull request closed without merging**